src/config/mod.rs

```diff
@@ -370,13 +371,14 @@ impl Config {
     }
 
     #[allow(dead_code)]
-    pub(super) fn from_toml(toml: &str, dir: &Path) -> Result<Config, String> {
-        Self::from_toml_for_style_edition(toml, dir, None, None, None)
+    pub(super) fn from_toml(toml: &str, dir: &Path,file_path : &Path) -> Result<Config, String> {
+        Self::from_toml_for_style_edition(toml, dir,file_path, None, None, None)
     }
 
     pub(crate) fn from_toml_for_style_edition(
         toml: &str,
         dir: &Path,
+        file_path: &Path,
```

Comment:
Same goes for this. We don't need to pass both the `dir` and `file_path`. We can derive the `dir` from the `file_path`.